app/kumactl/cmd/inspect/inspect_dataplanes.go

```diff
@@ -56,6 +56,8 @@ func newInspectDataplanesCmd(pctx *inspectContext) *cobra.Command {
 		},
 	}
 	cmd.PersistentFlags().StringToStringVarP(&ctx.tagsArgs.tags, "tag", "", map[string]string{}, "filter by tag in format of key=value. You can provide many tags")
+	cmd.PersistentFlags().StringToStringVarP(&ctx.tagsArgs.tags, "gateway", "", map[string]string{}, "filter gateway dataplanes")
```

Comment:
This should be a boolean not a map. Example:
```
kumactl inspect dataplanes --tag service=kong --gateway=true
```
or even
```
kumactl inspect dataplanes --tag service=kong --gateway
```

Reply:
gateway data plane has any service right, can we return all the gateway's by default and we can remove the --tag service=kong from it?

Reply:
and right now I am passing the `service=gateway` to the `tagsArgs` so no need to implement anything specific for `--gateway command`

Reply:
You don't have to specify service tag to filter gateway dataplanes. You can just as well do this
```
kumactl inspect dataplanes --gateway
```

Reply:
then I need to extend the `client.List` to accept one more argument, so that I can add the query param to filter gateway data planes since we are making an Http call to our service. Am I correct on this?

Reply:
correct